tfhe/src/integer/server_key/radix_parallel/tests.rs

```diff
@@ -743,6 +746,144 @@ where
     }
 }
 
+fn integer_default_scalar_bitand<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    let (cks, sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+
+    let mut clear;
+
+    for _ in 0..NB_TEST_SMALLER {
+        let clear_0 = rng.gen::<u64>() % modulus;
+        let clear_1 = rng.gen::<u64>() % modulus;
+
+        // encryption of an integer
+        let ctxt_0 = cks.encrypt(clear_0);
+
+        // Do with a small clear to check the way we avoid
+        // unecesseray work is correct
+        let ct_res = sks.scalar_bitand_parallelized(&ctxt_0, 1);
+        let dec_res: u64 = cks.decrypt(&ct_res);
+        assert_eq!(clear_0 & 1, dec_res);
+
+        let mut ct_res = sks.scalar_bitand_parallelized(&ctxt_0, clear_1);
+        assert!(ct_res.block_carries_are_empty());
+
+        clear = clear_0 & clear_1;
+
+        for _ in 0..NB_TEST_SMALLER {
+            let clear_2 = rng.gen::<u64>() % modulus;
+
+            let tmp = sks.scalar_bitand_parallelized(&ct_res, clear_2);
+            ct_res = sks.scalar_bitand_parallelized(&ct_res, clear_2);
+            assert!(ct_res.block_carries_are_empty());
+            assert_eq!(ct_res, tmp);
+            clear &= clear_2;
+
+            let dec_res: u64 = cks.decrypt(&ct_res);
+            assert_eq!(clear, dec_res);
+        }
+    }
+}
+
+fn integer_default_scalar_bitor<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    let (cks, sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    //RNG
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+
+    let mut clear;
+
+    for _ in 0..NB_TEST_SMALLER {
+        let clear_0 = rng.gen::<u64>() % modulus;
+        let clear_1 = rng.gen::<u64>() % modulus;
+
+        let ctxt_0 = cks.encrypt(clear_0);
+
+        // Do with a small clear to check the way we avoid
+        // unecesseray work is correct
+        let ct_res = sks.scalar_bitor_parallelized(&ctxt_0, 1);
+        let dec_res: u64 = cks.decrypt(&ct_res);
+        assert_eq!(clear_0 | 1, dec_res);
+
+        let mut ct_res = sks.scalar_bitor_parallelized(&ctxt_0, clear_1);
+        assert!(ct_res.block_carries_are_empty());
+        clear = (clear_0 | clear_1) % modulus;
```

Comment:
why is the modulus required here ?

Reply:
It should not I copy pasted this from somewhere else